app/com/linkedin/drelephant/tuning/obt/ParameterGenerateManagerOBTAlgoPSOIPSOImpl.java

```diff
@@ -18,43 +18,18 @@
 import org.apache.log4j.Logger;
 
 
-public class ParameterGenerateManagerOBTAlgoPSOIPSOImpl extends ParameterGenerateManagerOBTAlgoPSO {
+public abstract class ParameterGenerateManagerOBTAlgoPSOIPSOImpl<T extends ExecutionEngine>
+    extends ParameterGenerateManagerOBTAlgoPSO<T> {
 
   private final Logger logger = Logger.getLogger(getClass());
 
   enum UsageCounterSchema {USED_PHYSICAL_MEMORY, USED_VIRTUAL_MEMORY, USED_HEAP_MEMORY}
```

Comment:
Make it protected if you want to use it in subclasses instead of making it default access. If you using in test, mark it as @VisibleForTesting